uwdata/lyra
# Hints engine

Adds Hints functionality that listens to actions in redux and shows a box.
![hints](https://cloud.githubusercontent.com/assets/607541/14689738/727daad8-06fd-11e6-91cb-a067e5d96952.gif)

The Hints reducer contains any listeners for other app actions such as `MARK_ADD`. 
When those actions are observed, modify the "display" in the hints store to be a hints object.
( I store the actual hints text and objects in the hints folder to make things less cluttered) 

This is what the hints objects look like:

```
{
  title: 'I see you have added a RECT...',
  text: 'Maybe you\'d like to clear the whole canvas',
  action_text: 'DO IT. CLEAR IT OUT.',  // optional: button text
  action: require('../actions/sceneClear'), // optional: button onClick event
  action_props: {} // optional: any properties that need to get passed in the button callback
}
```

or for custom templates:

```
{
  template: require('../components/hints/custom-hint')
}
```

**Steps to functionally test this:**
-  Hints are added to the following actions: create scene, create rect, line, and symbol
- symbol - should have a custom template
- toggle the hints on and off in the settings modal (accessed via the footer)
  - after turning the hints off, click on RECT, SYMBOL or LINE to make sure they do not show up
